tmail-backend/webadmin/webadmin-mailbox/src/main/java/com/linagora/tmail/webadmin/cleanup/CleanupTrashService.java

```diff
@@ -54,54 +67,72 @@ public Mono<Task.Result> cleanupTrash(RunningOptions runningOptions) {
             .transform(ReactorUtils.<Username, Task.Result>throttle()
                 .elements(runningOptions.getUsersPerSecond())
                 .per(Duration.ofSeconds(1))
-                .forOperation(this::checkSettingAndCleanupTrashForSingleUser))
+                .forOperation(this::cleanupTrashForSingleUser))
             .reduce(Task.Result.COMPLETED, Task::combine)
             .onErrorResume(e -> {
                 LOGGER.error("Error while accessing users from repository", e);
                 return Mono.just(Task.Result.PARTIAL);
             });
     }
 
-    private Mono<Task.Result> checkSettingAndCleanupTrashForSingleUser(Username username) {
+    private Mono<Task.Result> cleanupTrashForSingleUser(Username username) {
         return Mono.from(jmapSettingsRepository.get(username))
-            .flatMap(jmapSettings -> checkSettingAndCleanupTrashForSingleUser(username, jmapSettings))
+            .flatMap(jmapSettings -> cleanupTrashForSingleUser(username, jmapSettings))
             .onErrorResume(e -> {
                 LOGGER.error("Error while cleaning trash for user {}", username, e);
                 return Mono.just(Task.Result.PARTIAL);
             });
     }
 
-    private Mono<Task.Result> checkSettingAndCleanupTrashForSingleUser(Username username, JmapSettings jmapSettings) {
-        return getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_ENABLED_SETTING)
-            .map(jmapSettingsValue -> Mono.just(jmapSettingsValue).filter(trashCleanupEnabled -> Boolean.parseBoolean(trashCleanupEnabled.value()))
-                .flatMap(trashCleanupEnabled -> cleanupTrashForSingleUser(username,
-                    getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_PERIOD_SETTING).orElse(new JmapSettings.JmapSettingsValue(MONTHLY_PERIOD)).value()))
-                .switchIfEmpty(Mono.just(Task.Result.COMPLETED)))
-            .orElse(Mono.just(Task.Result.COMPLETED));
+    private Mono<Task.Result> cleanupTrashForSingleUser(Username username, JmapSettings jmapSettings) {
+        return getTrashCleanupEnabledSetting(jmapSettings)
+            .filter(Boolean::booleanValue)
+            .flatMap(trashCleanupEnabled -> cleanupTrashForSingleUserWithSpecificPeriod(username, getTrashCleanupPeriodSetting(jmapSettings)))
+            .switchIfEmpty(Mono.just(Task.Result.COMPLETED));
+    }
+
+    private JmapSettingsValue getTrashCleanupPeriodSetting(JmapSettings jmapSettings) {
+        return getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_PERIOD_SETTING, MONTHLY_PERIOD);
+    }
+
+    private Mono<Boolean> getTrashCleanupEnabledSetting(JmapSettings jmapSettings) {
+        return Mono.fromCallable(() -> getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_ENABLED_SETTING)
+            .map(trashCleanupEnabled -> Boolean.parseBoolean(trashCleanupEnabled.value()))
+            .orElse(false));
```

Comment:
Should we abstract the setting key and default setting value in `JmapSettings` class?
Something like:
```scala
case class JmapSettings(settings: Map[JmapSettingsKey, JmapSettingsValue], state: UuidState) {
  def trashCleanupEnable(): Boolean =
    settings.get(JmapSettingsKey.liftOrThrow("trash.cleanup.enabled"))
      .map(value => stringToBoolean)
      .getOrElse(false)
}
```